### ISSUE TITLE
Support log search functionality in Pinot

### ISSUE THREAD

**Issue by ananthdurai** [apache/pinot#2798]
Pinot provides support to pick and choose to index for each column. Adding the support for log search indexing will help some of the use cases like error stack trace indexing etc.
Most of the finding the needle in the haystack use cases starts with a free text search and then drill down analysis. Adding support for the free text search can enable Pinot to address the extended use cases.

**Comment by kgopalakrishna** [comment]
Can you add more details on the type of operations (exact match, prefix match and boolean operations) that can be performed on these columns?. Do you need everything that lucene provides?

**Comment by ananthdurai** [comment]
Thank you @kishoreg. I think the scope of the search capability can be limited to the following, and I'm assuming the column `error_logs` is the full text indexed column/
1. prefix Query ( error_logs = "error*")
2. wildcard Query (error_logs = "apache*error")
3. regex Query ( error_logs = "error.*access")
4. Fuzzy Query (fuzzy_search(term="apache", fuzziness = 3))
5. Match Phrase Query ( error_logs = "error")
The proposal assumes that users may choose some column to be indexed as free text.

**Comment by mcvsubbu** [comment]
We do support regexp_like(). See https://pinot.readthedocs.io/en/latest/pql_examples.html

**Comment by ananthdurai** [comment]
TIL: Let me run some performance comparison study on it.

**Comment by ananthdurai** [comment]
@mayankshriv Yes, that is correct. Ideally, if Pinot can support multiple indexing formats, (log search and the columnar indexing) and the query engine intelligent enough to choose the indexes based on the query pattern.

**Comment by mayankshriv** [comment]
@ananthdurai Agreed, that would be a good feature to have. @sunithabeeram was experimenting with it with @kishoreg, will let them comment on how that is panning out.

**Comment by ananthdurai** [comment]
Adding a couple of links from DM for future references,
https://www.quora.com/What-is-the-algorithm-used-by-Lucenes-PrefixQuery
https://issues.apache.org/jira/browse/LUCENE-1606

**Comment by israbbani** [comment]
+1

**Comment by Jackie-Jiang** [comment]
Supported via Lucene text search
